testsuite/tests/lib-string/test_string.ml

```diff
@@ -40,6 +40,11 @@ let () =
   done
 ;;
 
+
+let () =
+  assert((Hashtbl.hash raw_string) = (String.hash raw_string))
+;;
```

Comment:
Should there be (or is there) a test that makes sure that the hashing function doesn't just return the same (e.g. 0) for all values?